docs/decisions/0006-role-of-xblock.rst

```diff
@@ -0,0 +1,130 @@
+that *look* dynamic were actually constrainted with unstated assumptions–maximum
+scores are assumed to be consistent across all user, courses are assumed to have
+a Course → Section → Subsection structure, etc.
+
+Reaching beyond these constraints is still possible, but the results are
+unpredictable. The LMS will still render unusual navigational structures if you
+import it directly via OLX, but mobile clients and analytics will likely break.
+You can define different max scores for different students on the same problem,
+but you will likely see bugs in various places where those scores are cached and
+displayed.
+
+Largely because of these issues, the XBlock runtime gradually shifted away from
+the all-encompassing role it once played in courseware:
+
+* Grading and Scheduling became distinct subsystems with their own data models
+  and APIs. Instead of these systems synchronously pulling data from XBlock at
+  query time, we push XBlock data into these systems and query them using their
+  own APIs. This has vastly increased the operational stability of the platform.
+* LTI support has been extracted from ``LtiConsumerXBlock``, so that LTI
+  integration can happen outside the XBlock runtime.
+* There is active planning to add limited QTI support to Open edX.
+* The ``learning_sequence`` application is being developed to take over Section
+  and Sequence level navigation, decoupling those concepts from the XBlock
+  runtime in the LMS.
+* There is a strong interest in creating more adaptive and composable courseware
+  experiences which are difficult to do within the constraints of the XBlock
+  runtime.
+* There have been early conversations around what a successor to XBlock might
+  look like, though no development work has begun.
+
+At the same time, it's important to note that there are hundreds of XBlocks in
+existence and a vast quantity of valuable content that has been written for
+those XBlocks. Preserving the viability of this content is critical to the
+success of Open edX.
+
+
+Decision
+========
+
+XBlocks have provided us a fantastic source of innovative content, at the cost
+of many operational issues. We'll proceed in a way that allows us to preserve
+the most valuable parts of the XBlock framework, while developing a more
+performant, XBlock-agnostic layer of extensible LMS APIs.
+
+In concrete terms:
+
+* XBlocks will continue to be supported at the Unit and individual module level
+  (e.g. ProblemBlock, VideoBlock). XBlocks will execute in a container that only
+  has their Unit (VerticalBlock). They will be allowed to query sibling blocks,
+  but will no longer be allowed to freely query their ancestors to get at
+  content in other units, sequences, or the root Course block.
+* Responsibility for higher level navigation across collections of content (i.e.
+  Sequences, Sections, and Courses) will move to dedicated applications that
+  will have their own, XBlock-agnostic data models and APIs.
```

Comment:
Is this for LMS only or for Studio as well (assuming the definition of Studio from https://github.com/edx/edx-platform/pull/24463#discussion_r456423808)?

Reply:
LMS is definitely the priority. I expect the transition period will involve keeping things as-is in Studio for now, and doing data transformation at the time of publish, per the guidelines in #24463. I would expect that eventually Studio would follow, or a new system would start publishing course data, but the timeline and approach for that is pretty fuzzy in my head at the moment.